src/ast/irbuilderbpf.cpp

```diff
@@ -1496,6 +1549,24 @@ void IRBuilderBPF::CreatePerfEventOutput(Value *ctx,
                    loc);
 }
 
+void IRBuilderBPF::CreateDebugOutput(std::string fmt_str,
```

Comment:
Why is this not used for `debugf`, too? IIUC, if you now want to dump `debugf` output to stderr, you always need to add `BPFTRACE_DEBUG_OUTPUT` substring, is that intentional? It seems so to me when I'm looking at the new runtime test.

Reply:
> IIUC, if you now want to dump debugf output to stderr, you always need to add BPFTRACE_DEBUG_OUTPUT substring, is that intentional?

Yes, but I'm not quite sure about it.

The current setting is:
* the built-in `debugf` is for the convenience of our bpftrace development, never for end users. This is implemented just like `printf`, we can customize its output. The drawback is we have to look into the trace-pipe by ourselves.
* the BPFTRACE_DEBUG_OUTPUT is something like tracepoints that we intentionally left in the bpftrace codes, to provide diagnostic information under unexpected circumstances. These messages should be well-formatted, easily-understandable.

Anyway, I'm okay with using the `CreateDebugOutput` to achieve the built-in `debugf` if you say so.

Reply:
Right, makes sense. Let's keep it as it is now.